airbyte-integrations/connectors/source-yandex-metrica/source_yandex_metrica/streams.py

```diff
@@ -0,0 +1,345 @@
+    ) -> MutableMapping[str, Any]:
+        return super().request_params(stream_state=stream_state)
+
+    @property
+    def raise_on_http_errors(self) -> bool:
+        return False
+
+    def parse_response(
+        self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        yield from self.fetch_records(response, stream_state)
+
+# Substreams
+class Create(YandexMetricaStream):
+    primary_key = None
+
+    def __init__(self, counter_id: str, params: dict, **kwargs):
+        super().__init__(counter_id, params, **kwargs)
+
+    @property
+    def http_method(self) -> str:
+        return "POST"    
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"{self.counter_id}/logrequests"
+    
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+
+        return {
+            'date1': self.params['start_date'],
+            'date2': self.params['end_date'],
+            'source': self.params['source'],
+            'fields': self.params['fields']
+        }
+
+    def parse_response(
+        self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        data = response.json()
+        yield data
+class Check(YandexMetricaStream):
+    primary_key = None
+
+    def __init__(self, counter_id: str, params: dict, logrequest_id: int, **kwargs):
+        self.logrequest_id = logrequest_id
+        super().__init__(counter_id, params, **kwargs)
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"{self.counter_id}/logrequest/{self.logrequest_id}"
+
+    @property
+    def max_retries(self) -> Union[int, None]:
+        return 240
+        
+    def should_retry(self, response: requests.Response) -> bool:
+        data = response.json()
+        return data["log_request"]["status"] != YMStatus.PROCESSED
+    
+    def backoff_time(self, response: requests.Response) -> Optional[float]:
+        return 30
+
+    def parse_response(
+        self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        data = response.json()
+        yield data
+
+class Download(YandexMetricaStream):
```

Comment:
Please add link to Yandex endpoint api documentation

Reply:
Added docstrings to streams.